Mathlib/Combinatorics/SimpleGraph/ConcreteColorings.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2023 Iván Renison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Iván Renison
+-/
+import Mathlib.Combinatorics.SimpleGraph.Coloring
+import Mathlib.Combinatorics.SimpleGraph.Hasse
+import Mathlib.Data.ZMod.Basic
+
+/-!
+# Concrete colorings of common graphs
+
+This file defines colorings for some common graphs
+
+## Main declarations
+
+* `SimpleGraph.pathGraph.bicoloring`: Bicoloring of a path graph
+-/
+
+theorem suc_mod2_neq (m : ℕ) : (m % 2 == 0) ≠ ((m + 1) % 2 == 0) := by
+  intro h
+  rw [Bool.eq_iff_eq_true_iff, beq_iff_eq] at h
+  simp only [beq_iff_eq, ←Nat.even_iff, Nat.even_add_one] at h
+  exact not_iff_self h.symm
+
+/-- Bicoloring of a path graph -/
+def SimpleGraph.pathGraph.bicoloring (n : ℕ) :
+    SimpleGraph.Coloring (pathGraph n) Bool :=
+  Coloring.mk (fun u ↦ u.val % 2 == 0)
+    (by
+      intro u v h
+      simp
+      have : u ⋖ v ∨ v ⋖ u := by simp_all [pathGraph]
+      have : u.val ⋖ v.val ∨ v.val ⋖ u.val := by simp_all [Fin.coe_covby_iff]
+      have : u.val + 1 = v.val ∨ v.val + 1 = u.val := by simp_all [Nat.covby_iff_succ_eq]
+      match this with
+        | Or.inl h' =>
+          rw [←h']
+          exact suc_mod2_neq u
+        | Or.inr h' =>
+          rw [←h']
+          exact (suc_mod2_neq v).symm)
```

Comment:
Here's a shorter proof by using a lemma and some mathlib tactics. Could you add this lemma to right after `pathGraph`?
```suggestion
theorem pathGraph_adj {n : ℕ} {u v : Fin n} :
    (pathGraph n).Adj u v ↔ u.val + 1 = v.val ∨ v.val + 1 = u.val := by
  simp only [pathGraph, hasse]
  simp_rw [← Fin.coe_covby_iff, Nat.covby_iff_succ_eq]

/-- Bicoloring of a path graph -/
def pathGraph.bicoloring (n : ℕ) :
    Coloring (pathGraph n) Bool :=
  Coloring.mk (fun u ↦ u.val % 2 = 0) <| by
    intro u v
    rw [pathGraph_adj]
    rintro (h | h) <;> simp [← h, not_iff, succ_mod_two_eq_zero_iff]
```

Style comments:
- Do not have "non-terminal simps," which are simps that do not close a goal. That `simp` before the `have` is not allowed -- we use `simp?` to turn them into a `simp only`, which is allowed.
- We generally don't do `simp_all`. Usually there's something more precise.
- `match` is ok, but using `obtain` or `rintro` tends to be cleaner. Inside tactics, you'll often find `cases` instead of `match`, which has similar syntax.